src/jit/flowgraph.cpp

```diff
@@ -17353,10 +17353,20 @@ BasicBlock*         Compiler::fgAddCodeRef(BasicBlock*      srcBlk,
                                            SpecialCodeKind  kind,
                                            unsigned         stkDepth)
 {
-    /* For debuggable code, genJumpToThrowHlpBlk() will generate the 'throw'
-       code inline. It has to be kept consistent with fgAddCodeRef() */
+    // For debuggable code, genJumpToThrowHlpBlk() will generate the 'throw'
+    // code inline. It has to be kept consistent with fgAddCodeRef()
     if (opts.compDbgCode)
-        return NULL;
+    {
+        if (compIsAmd64ABI() && !compIsUnixAmd64ABI())
```

Comment:
Although I like these new helpers compIs---ABI I don't think that we need to use them for this case.
 
Since the Codegen for this is the same for both Unix and Windows and all other platforms 
We should just record this fact  under compDbgCode using 

    compDbgCallsThrowHelper = true; 

Or even unconditionally using

    compUsesThrowHelper = true;